### PR TITLE
REF: move EA wrapping/unwrapping to indexes.extensions

### PR DESCRIPTION
**Pull request by jbrockmendel**
Re-use the comparison method wrapper in CategoricalIndex.

### CODE REVIEW

**Review by TomAugspurger — APPROVED** [review]
Looks good.

File: pandas/core/indexes/extension.py [hunk 1]
```diff
@@ -76,3 +82,73 @@ def wrapper(cls):
         return cls
 
     return wrapper
+
+
+def make_wrapped_comparison_op(opname):
```
Comment (jreback) [inline pandas/core/indexes/extension.py]: at some point should type these

**Review by jreback — APPROVED** [review]
lgtm. some followon comments

**Comment by jreback** [comment]
thanks

**Pull request merged**